# Question: How do the ticket prices correlate to survival rates?
# Observational study: 
# I have performed a retrospective census on a data set of passengers on the titanic. This study may produce bias due to the varying ticket prices and high mean value. 

In [4]:
!pip install pandas

Defaulting to user installation because normal site-packages is not writeable
Looking in links: /usr/share/pip-wheels


In [3]:
import pandas as pd

In [5]:
df = pd.read_csv("usdticketprices2025.csv")
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3 entries, 0 to 2
Data columns (total 4 columns):
 #   Column                       Non-Null Count  Dtype
---  ------                       --------------  -----
 0   Pclass                       3 non-null      int64
 1   MedianFare_1912_GBP          3 non-null      int64
 2   AdjustedMedianFare_2025_GBP  3 non-null      int64
 3   AdjustedMedianFare_2025_USD  3 non-null      int64
dtypes: int64(4)
memory usage: 224.0 bytes


In [29]:
df = pd.read_csv("Titanic-Dataset.csv")
df.info()
num_samples = df.shape[0]
print("Number of samples:", num_samples)
print(df.iloc[6])

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    object 
 4   Sex          891 non-null    object 
 5   Age          714 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    object 
 9   Fare         891 non-null    float64
 10  Cabin        204 non-null    object 
 11  Embarked     889 non-null    object 
dtypes: float64(2), int64(5), object(5)
memory usage: 83.7+ KB
Number of samples: 891
PassengerId                          7
Survived                             0
Pclass                               1
Name           McCarthy, Mr. Timothy J
Sex                               male
Age                       

In [11]:
df = pd.read_csv("titanic.csv")
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 887 entries, 0 to 886
Data columns (total 8 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   Survived                 887 non-null    int64  
 1   Pclass                   887 non-null    int64  
 2   Name                     887 non-null    object 
 3   Sex                      887 non-null    object 
 4   Age                      887 non-null    float64
 5   Siblings/Spouses Aboard  887 non-null    int64  
 6   Parents/Children Aboard  887 non-null    int64  
 7   Fare                     887 non-null    float64
dtypes: float64(2), int64(4), object(2)
memory usage: 55.6+ KB


In [9]:
df = pd.read_csv("Titanic-Dataset.csv")
mean_value = 0
for value in df["Fare"]:
    mean_value = mean_value + value
mean_value = mean_value/df.shape[0]
print("Mean: ", mean_value)
count_rs = 0
count_rd = 0
count_ps = 0
count_pd = 0
for row in df.itertuples(index=True):
    if(row.Fare >= mean_value):
        if(row.Survived == 1):
            count_rs += 1
        else:
            count_rd += 1
    else:
        if(row.Survived == 1):
            count_ps += 1
        else:
            count_pd += 1
print(count_rs)
print(count_rd)
print(count_ps)
print(count_pd)

Mean:  32.2042079685746
126
85
216
464


# Out of all 211 passengers who spent more than 32 dollars on their ticket, 126 survived and 85 died, and out of all 680 passengers who spent less than $32 dollars on their ticket, 216 survived and 464 died. This data fits the pattern that the more you spent on your ticket, the higher your chances of survival were.

In [8]:
df = pd.read_csv("Titanic-Dataset.csv")
median_value = df["Fare"].median()
print("Median:", median_value)
count_rs = 0
count_rd = 0
count_ps = 0
count_pd = 0
for row in df.itertuples(index=True):
    if(row.Fare >= median_value):
        if(row.Survived == 1):
            count_rs += 1
        else:
            count_rd += 1
    else:
        if(row.Survived == 1):
            count_ps += 1
        else:
            count_pd += 1
print(count_rs)
print(count_rd)
print(count_ps)
print(count_pd)

Median: 14.4542
231
220
111
329
